app/helpers/form_helpers.py

```diff
@@ -0,0 +1,88 @@
+from datetime import timedelta
+from functools import wraps
+
+from flask import abort, current_app, render_template, request, Response, session
+from wtforms import Form
+from wtforms.csrf.core import CSRF
+from wtforms.csrf.session import SessionCSRF
+
+
+class FakeCsrf(CSRF):
+    """
+    For testing purposes only.
+    """
+
+    valid_token = 'valid_fake_csrf_token'
+
+    def generate_csrf_token(self, csrf_token):
+        return self.valid_token
+
+    def validate_csrf_token(self, form, field):
+        if field.data != self.valid_token:
+            raise ValueError('Invalid (fake) CSRF token')
+
+
+class DmForm(Form):
+
+    class Meta:
+        csrf = True
+        csrf_class = SessionCSRF
+        csrf_secret = None
+        csrf_time_limit = None
+
+        @property
+        def csrf_context(self):
+            return session
+
+    def __init__(self, *args, **kwargs):
+        if current_app.config['CSRF_ENABLED']:
+            self.Meta.csrf_secret = current_app.config['SECRET_KEY']
+            self.Meta.csrf_time_limit = timedelta(seconds=current_app.config['CSRF_TIME_LIMIT'])
+        if current_app.config.get('CSRF_FAKED', False):
+            self.Meta.csrf_class = FakeCsrf
+        else:
+            # FIXME: deprecated
+            self.Meta.csrf = False
+            self.Meta.csrf_class = None
+        super(DmForm, self).__init__(*args, **kwargs)
+
+
+def render_template_with_csrf(template_name, status_code=200, **kwargs):
+    if 'form' not in kwargs:
+        kwargs['form'] = DmForm()
+    response = Response(render_template(template_name, **kwargs))
+
+    # CSRF tokens are user-specific, even if the user isn't logged in
+    response.cache_control.private = True
+
+    max_age = current_app.config['DM_DEFAULT_CACHE_MAX_AGE']
+    max_age = min(max_age, current_app.config.get('CSRF_TIME_LIMIT', max_age))
```

Comment:
The time limit itself is handled by the third-party CSRF token library.  This line just makes sure that we don't ask browsers to cache the page longer than the token inside it is valid.